src/optimizer/plan_generation.c

```diff
@@ -1352,6 +1356,24 @@ make_pred_from_plan (QO_ENV * env, QO_PLAN * plan, PT_NODE ** key_predp, PT_NODE
   /* make predicate list for key filter */
   if (key_predp != NULL)
     {
+      if (qo_index_infop->need_copy_multi_range_term != -1 && qo_index_infop->need_copy_to_sarg_term)
```

Comment:
might need to check if qo_index_infop is null ?

Reply:
fixed